base/system/winlogon/notify.c

```diff
@@ -356,23 +357,38 @@ CallNotificationDll(
                                    &dwType,
                                    (PBYTE)szFuncBuffer,
                                    &dwSize);
+
+        RegCloseKey(hDllKey);
     }
 
-    if (dwError == ERROR_SUCCESS)
+    if (dwError != ERROR_SUCCESS)
+        return;
+
+    _SEH2_TRY
     {
-        hModule = LoadLibraryW(NotificationDll->pszDllName);
-        if (hModule != NULL)
+        _SEH2_TRY
```

Comment:
Wow, nested SEH2, now I've seen everything :smile:

Reply:
Yes, this exists (see also msdn). That's when you want to have both the "finally" and the "except" functionality of SEH at the same time.

Reply:
It should not be needed around LoadLibraryW, because that itself already uses SEH.
https://git.reactos.org/?p=reactos.git;a=blob;f=dll/ntdll/ldr/ldrapi.c;hb=cd2d284142c1fcaefc9759412c51c6d2fb2ca461#l356
and
https://git.reactos.org/?p=reactos.git;a=blob;f=dll/ntdll/ldr/ldrutils.c;hb=cd2d284142c1fcaefc9759412c51c6d2fb2ca461#l2497

Reply:
Fixed!